src/languages/java.md

```diff
@@ -1,3 +1,64 @@
 # Java
 
-Java support is in Alpha and not generally available. Contact sales [https://platform.sh/contact](https://platform.sh/contact) if you are interested in running Java based applications on Platform.sh.
+Java is a general-purpose programming language, and one of the most popular in the world today. Platform.sh supports Java runtimes that can be used with build management tools such as Gradle, Maven, and Ant.
+
+
+## Supported versions
+
+### OpenJDK versions:
+
+* 8
+* 11
+* 12
+
+
+To select a Java version, specify a `type` such as `java:11`:
+
+```yaml
+# .platform.app.yaml
+type: "java:11"
+```
+
+
+## Support libraries
+
+While it is possible to read the environment directly from your application, it is generally easier and more robust to use the [`platformsh/config-reader`](https://github.com/platformsh/config-reader-java) which handles decoding of service credential information for you.
+
+## Support build automation
+
+Platform.sh supports the most common project management tools in the Java ecosystem, including: 
+
+* [Gradle](https://gradle.org/)
+* [Maven](https://maven.apache.org/)
+* [Ant](https://ant.apache.org/)
+
+
+
+## Accessing services
+
+To access various [services](/configuration/services.md) with Java, see the following examples.  The individual service pages have more information on configuring each service.
+
+
+
+{% language name="MongoDB", type="java", url="https://examples.docs.platform.sh/java/mongodb" -%}
```

Comment:
The first tag needs to be called "codetabs".  Only the subsequent ones are called "language".

Reply:
thank you